src/inferencer/typeEnvironment.ts

```diff
@@ -23,28 +24,59 @@ export function updateTypeEnvironment(program: es.Program) {
   function updateForConstantDeclaration(
     constantDeclaration: TypeAnnotatedNode<es.VariableDeclaration>
   ) {
-    // e.g. Given: const x^T1 = 1^T2, Set: Γ[ x ← T2 ]
-    const iden = constantDeclaration.declarations[0].id as TypeAnnotatedNode<es.Identifier>
-    const idenName = iden.name
-
-    const value = constantDeclaration.declarations[0].init as TypeAnnotatedNode<es.Node> // use es.Node because rhs could be any value/expression
-    const valueTypeVariable = value.typeVariable as Variable
-
-    if (idenName !== undefined && valueTypeVariable !== undefined) {
-      globalTypeEnvironment.set(idenName, {
-        types: [valueTypeVariable]
-      })
+    // First, check if it is a function definition
+    if (
+      (constantDeclaration.declarations[0].init as TypeAnnotatedNode<es.Node>).type ===
+      'ArrowFunctionExpression'
+    ) {
+      // If so, handle separately
+      updateForFunctionDefinition(constantDeclaration)
+    } else {
+      // Otherwise, should just be literals
+      // e.g. Given: const x^T1 = 1^T2, Set: Γ[ x ← T2 ]
+      const iden = constantDeclaration.declarations[0].id as TypeAnnotatedNode<es.Identifier>
+      const idenName = iden.name
+
+      const value = constantDeclaration.declarations[0].init as TypeAnnotatedNode<es.Node> // use es.Node because rhs could be any value/expression
+      const valueTypeVariable = value.typeVariable as Variable
+
+      if (idenName !== undefined && valueTypeVariable !== undefined) {
+        globalTypeEnvironment.set(idenName, {
+          types: [valueTypeVariable]
+        })
+      }
     }
   }
 
+  function updateForFunctionDefinition(
+    constantDeclaration: TypeAnnotatedNode<es.VariableDeclaration>
```

Comment:
If we are updating function definitions like this:
```
y => y;
```
Should it take in a `TypeAnnotatedNode<es.ArrowFunctionExpression>` instead of a VariableDeclaration?

Reply:
Right, forgot it could come w/o a const decl. Let me add this part in

Reply:
Noted